python/tank/util/metrics.py

```diff
@@ -591,6 +725,12 @@ def log(cls, group, name, properties=None, log_once=False, bundle=None):
                 properties = base_properties
         # else we won't get base properties
 
+        # Now add basic platform information to the metric properties
+        if properties:
```

Comment:
Earlier in the method, you could say `properties = properties or {}` and then avoid all the `if properties` checks.